### ISSUE TITLE
tf.reduce_sum not consistent with documentation when axis==[ ]

### ISSUE THREAD

**Issue by tdanyluk** [tensorflow/tensorflow#19498]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**:  3.5.2
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:  9.0.176 / 7.0.5
- **GPU model and memory**: GeForce GTX 1050 Ti/PCIe/SSE2 4GB
- **Exact command to reproduce**: custom script

### Describe the problem
documentation says:
If axis has no entries, all dimensions are reduced, and a tensor with a single element is returned.

In practice:
If axis == []: nothing is reduced (the output is the input itself)

### Source code / logs
import tensorflow as tf

constant = tf.constant([[1., 2.], [3., 4.]])
reduce = tf.reduce_sum(constant, axis=[])

with tf.Session() as sess:
    print(sess.run(reduce))

**Comment by yongtang** [comment]
The current behavior of tensorflow matches numpy:
```
# python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> x = np.asarray([[1., 2.], [3., 4.]])
>>> np.sum(x, ())
array([[1., 2.],
       [3., 4.]])
>>> 
```

So I think this is a documentation (not implementation) issue. Created a PR #19509 for the documentation fix.
